### PR TITLE
[no sq] Unnamed collection of fixes

### PR DESCRIPTION
**Pull request by sfan5**
commit 1: on my system these values are always under 0.0xx milliseconds, so it isn't really useful to measure them using this precision.
commit 2: helps with debugging #14544 and possible future similar issues
commit 3: this has been broken for 7 years (at least)
the rest: I got sidetracked

### CODE REVIEW

File: src/servermap.cpp [hunk 1]
```diff
@@ -639,8 +639,7 @@ void ServerMap::loadBlock(std::string *blob, v3s16 p3d, MapSector *sector, bool
 	try {
 		std::istringstream is(*blob, std::ios_base::binary);
 
-		u8 version = SER_FMT_VER_INVALID;
-		is.read((char*)&version, 1);
+		u8 version = readU8(is);
```
Comment (grorp) [inline src/servermap.cpp:642]: So does EOF also result in the `is.fail()` condition being triggered?

Comment (sfan5) [reply src/servermap.cpp:642]: No idea, do you know?

Comment (grorp) [reply src/servermap.cpp:642]: Alright then.

**Review by grorp — APPROVED** [review]
(not tested)

**Pull request merged**